pkg/core/storelimit/limit.go

```diff
@@ -33,11 +33,21 @@ const (
 )
 
 // StoreLimit is an interface to control the operator rate of store
+// TODO: add a method to control the rate of store
+// the normal control flow is:
+// 1. check the store limit with Available in checker or scheduler.
+// 2. check the store limit with Available in operator controller again.
+// the different between 1 and 2 is that 1 maybe not use the operator level.
+// 3. take the cost of operator with Take in operator controller.
+// 4. ack will put back the cost into the limit for the next waiting operator after the operator is finished.
+// the cost is the operator influence, so the influence should be same in the life of the operator.
 type StoreLimit interface {
 	// Available returns true if the store can accept the operator
 	Available(cost int64, typ Type, level constant.PriorityLevel) bool
 	// Take takes the cost of the operator, it returns false if the store can't accept any operators.
 	Take(count int64, typ Type, level constant.PriorityLevel) bool
 	// Reset resets the store limit
 	Reset(rate float64, typ Type)
+	// Ack put back the cost into the limit for the next waiting operator after the operator is finished.
```

Comment:
Suggest adding comments to explain this kind of interface is only used by snapshot limit.

Reply:
fixed it